packages/core/src/utils/FetchUtil.ts

```diff
@@ -14,6 +14,19 @@ interface PostArguments extends RequestArguments {
   body?: Record<string, unknown>
 }
 
+async function fetchData(...args: Parameters<typeof fetch>) {
+  const response = await fetch(...args)
+  if (!response.ok) {
+    // Create error object and reject if not a 2xx response code
+    const err = new Error(`HTTP status code: ${response.status}`, {
```

Comment:
Why do we do this? Is it for formalizing the errors? Are we sure it's not breaking any error handling of blockchain calls?

Reply:
FetchUtil was not throwing when receiving failed requests (neither 500 nor 400)
Wondering how this was working for you on swaps controller for instance 🤔